linux/scripts/deb-packaging.sh

```diff
@@ -50,19 +56,162 @@ source_action() {
 }
 
+output_log() {
+  echo "$1" >&2
+  builder_echo "$1"
+}
+
+output_ok() {
+  echo ":heavy_check_mark: $1" >&2
+  builder_echo green "OK: $1"
+}
+
+output_warning() {
+  echo ":warning: $1" >&2
+  builder_echo warning "WARNING: $1"
+}
+
+output_error() {
+  echo ":x: $1" >&2
+  builder_echo error "ERROR: $1"
+}
+
+check_api_not_changed() {
+  # Checks that the API did not change compared to what's documented in the .symbols file
+  tmpDir=$(mktemp -d)
+  # shellcheck disable=SC2064
+  trap "rm -rf \"${tmpDir}\"" ERR
+  dpkg -x "${BIN_PKG}" "${tmpDir}"
+  cd debian
+  dpkg-gensymbols -v"${VERSION}" -p"${PKG_NAME}" -e"${tmpDir}"/usr/lib/x86_64-linux-gnu/"${LIB_NAME}".so* -O"${PKG_NAME}.symbols" -c4
+  output_ok "${LIB_NAME} API didn't change"
+  cd "${REPO_ROOT}/linux"
+  rm -rf "${tmpDir}"
+  trap ERR
+}
+
+#
+# Compare the SHA of the base and head commits for changes to the .symbols file
+#
+is_symbols_file_changed() {
+  local CHANGED_REF CHANGED_BASE
+  CHANGED_REF=$(git rev-parse "${GIT_SHA}":"linux/debian/${PKG_NAME}.symbols")
+  CHANGED_BASE=$(git rev-parse "${GIT_BASE}":"linux/debian/${PKG_NAME}.symbols")
+  if [[ "${CHANGED_REF}" == "${CHANGED_BASE}" ]]; then
+    return 1
+  fi
+  return 0
+}
+
+check_updated_version_number() {
+  # Checks that the package version number got updated in the .symbols file if it got changed
+  # shellcheck disable=SC2310
+  if is_symbols_file_changed; then
+    # .symbols file changed, now check if the package version got updated as well
+    # Note: We don't check that ALL changes in that file have an updated package version -
+    # we hope this gets flagged in code review.
+    # Note: This version number check may not match the actual released version, if the branch
+    # is out of date when it is merged to the release branch (master/beta/stable-x.y). If this
+    # is considered important, then make sure the branch is up to date, and wait for test
+    # builds to complete, before merging.
+    if ! git log -p -1 -- "debian/${PKG_NAME}.symbols" | grep -q "${VERSION}"; then
+      output_error "${PKG_NAME}.symbols file got changed without changing the package version number of the symbol"
+      EXIT_CODE=1
+    else
+      output_ok "${PKG_NAME}.symbols file got updated with package version number"
+    fi
+  else
+    output_ok "${PKG_NAME}.symbols file didn't change"
+  fi
+}
+
+get_api_version_in_symbols_file() {
+  # Extract 1 from "libkeymancore.so.1 libkeymancore #MINVER#"
+  local firstline
+  firstline="$(head -1 "debian/${PKG_NAME}.symbols")"
+  firstline="${firstline#"${PKG_NAME}".so.}"
+  firstline="${firstline%% *}"
+  echo "${firstline}"
+}
+
+is_api_version_updated() {
+  local NEW_VERSION OLD_VERSION
+  git checkout "${GIT_SHA}" -- "debian/${PKG_NAME}.symbols"
+  NEW_VERSION=$(get_api_version_in_symbols_file)
+  git checkout "${GIT_BASE}" -- "debian/${PKG_NAME}.symbols"
+  OLD_VERSION=$(get_api_version_in_symbols_file)
+  git checkout "${GIT_SHA}" -- "debian/${PKG_NAME}.symbols"
```

Comment:
We could reduce the number of checkouts by switching order of operations here, but it's not terribly important.